### PR TITLE
Query and update

### PR DESCRIPTION
**Pull request by J-C-S-V**
Hello dear reviewer 👋, I hope you're having a nice day today 😊

This is what I did on this branch:
- Add a column species of type string to your animals table.
- Insert more data into the table
- Write queries to get specific data

These are the screenshots for the queries:

Delete all animals born after Jan 1st, 2022.
Delete all animals born after Jan 1st, 2022

How many animals are there
How many animals are there

How many animals have never tried to escape
How many animals have never tried to escape

Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
Now, take a deep breath and  Inside a transaction delete all records in the animals table, then roll back the transaction

Update all animals' weight to be their weight multiplied by -1.
Update all animals' weight to be their weight multiplied by -1

Update all animals' weights that are negative to be their weight multiplied by -1.
Update all animals' weights that are negative to be their weight multiplied by -1

Update the animals table by setting the species column to digimon for all animals that have a name ending in mon
Update the animals table by setting the species column to digimon for all animals that have a name ending in mon

Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
Update the animals table by setting the species column to pokemon for all animals that don't have species already set

What is the average number of escape attempts per animal type of those born between 1990 and 2000
What is the average number of escape attempts per animal type of those born between 1990 and 2000

What is the average weight of animals
What is the average weight of animals

What is the minimum and maximum weight of each type of animal
What is the minimum and maximum weight of each type of animal

Who escapes the most, neutered or not neutered animals
Who escapes the most, neutered or not neutered animals

Regards! ✨

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
+	('Gabumon', '2018-11-15', 2, true, 8),
+	('Pikachu', '2021-01-07', 1, false, 15.04),
+	('Devimon', '2017-05-12', 5, true, 11),
+  ('Charmander', '2020-02-08', 0, false, 11),
+  ('Plantmon', '2021-11-15', 2, true, 5.7),
+  ('Squirtle', '1993-04-02', 3, false, 12.13),
+  ('Angemon', '2005-06-12', 1, true, 45),
+  ('Boarmon', '2005-06-07', 7, true, 20.4),
+  ('Blossom', '1998-10-13', 3, true, 17),
+  ('Ditto', '2022-05-14', 4, true, 22);
```
Comment (Owusu-Desmond) [inline data.sql]: - [x] Please from the project requirement, You should Insert the following data:
  - Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
  - Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
  - Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
  - Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
  - Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
  - Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
  - Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.

The data to be inserted is 7 rows but you are inserting more than 7 and besides you are inserting wrong data as compared to information about the data to be inserted above :blush: For instance _Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg_ the data for the `weight _kg` column should be -11.0 but you inserted it as 11.0 :smile:

File: queries.sql [hunk 2]
```diff
@@ -16,10 +16,51 @@ SELECT name, escape_attempts FROM animals
...
-WHERE neutered = true
+WHERE neutered = true;
 
 SELECT * FROM animals
 WHERE name NOT IN ('Gabumon');
 
 SELECT * FROM animals
 WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
```
Comment (Owusu-Desmond) [inline queries.sql]: - [x] Please you missed this query, from the project requirement, Inside a transaction update the `animals` table by setting the `species` column to `unspecified`. Verify that the change was made. Then roll back the change and verify that the `species` columns went back to the state before the transaction. :pray:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @J-C-S-V  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (J-C-S-V) [reply data.sql]: Hi @Owusu-Desmond,

These first 4 animals are comming from the previous exercise:

`('Agumon', '2020-02-03', 0, true, 10.23),
	('Gabumon', '2018-11-15', 2, true, 8),
	('Pikachu', '2021-01-07', 1, false, 15.04),
	('Devimon', '2017-05-12', 5, true, 11),`

Do you want me to remove those from the `data.sql` file?

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @J-C-S-V       , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] No linter errors
- [x] Query updates correctly implemented💯
 

 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**